pkg/operator/reports_test.go

```diff
@@ -92,3 +101,199 @@ func TestGetNextReportPeriod(t *testing.T) {
+		expectFinished bool
+	}{
+		{
+			name:           "new report returns false",
+			report:         testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectFinished: false,
+		},
+		{
+			name: "finished status on run-once report returns true",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, nil, false),
+			expectFinished: true,
+		},
+		{
+			name: "unset reportingEnd returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, nil, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "reportingEnd > lastReportTime returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 0, 0)},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "reportingEnd < lastReportTime returns true",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 2, 0)},
+			}, schedule, false),
+			expectFinished: true,
+		},
+		{
+			name: "when status running is false and reason is Scheduled return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is false and reason is InvalidReport return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is Scheduled return false",
```

Comment:
Nit: could you put this one immediately below the case with name `name: "when status running is false and reason is Scheduled return false"`. Having them closer visually should hopefully help make it easier to distinguish them from each other.